credentials/test/gen-test-attestation-certs.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Script that can be used to generate CHIP device attestation certificates
+# for testing purposes.
+# The script expects the path to the chip-cert tool binary as an input argument.
+#
+# Usage example when the script is run from the CHIP SDK root directory:
+#     ./credentials/test/gen-test-attestation-certs.sh ./out/debug/standalone/chip-cert
+#
+# The result will be stored in:
+#     credentials/test/attestation
+#
+# If the intention is to re-generate a new set of attestation certificates that
+# replace the already present ones then it is recommended to remove current certificates:
+#     rm credentials/test/attestation/*
+#
+
+set -e
+
+here=${0%/*}
+
+dest_dir="$here/attestation"
+
+mkdir -p "$dest_dir"
+
+if [ $# == 1 ]; then
+    chip_cert_tool=$1
+else
+    echo "Error: Please specify exactly one input argument; the path to the chip-cert tool binary"
+    exit
+fi
+
+cert_valid_from="2021-06-28 14:23:43"
+cert_lifetime=7305
+
+paa_vids=(FFF1 FFF2)
+pai_pids=(ABC0 ABC1)
+dac_ids=(0001 0002 0003 0004 0005 0006 0007 0008 0009 000A 000B 000C 000D 000E 000F)
+
+for vid in "${paa_vids[@]}"; do
+    paa_key_file="$dest_dir/Chip-Test-PAA-$vid-Key"
+    paa_cert_file="$dest_dir/Chip-Test-PAA-$vid-Cert"
```

Comment:
What would be best is to generate the test PAAs only once, and include the gen code in comments in here. Then we could start "officially" storing those dev PAAs and including them in DCL soon. For the PAI and DAC, it's OK to generate them.

I think overall we must avoid these test PAAs ever changing, so as to avoid having to rotate them and deal with possible commissioning tests downtime in the future.

Similarly the *dev* PAAs here should have infinite validity period.

Reply:
This script is not a tool but just a reference example of how these attestation certificates were generated. The intention here is that this script shouldn't be used as is to re-generate certificates that already present. I think it is OK to leave it as is and for any new certificates that should be added this script will be manually updated according to the requirements.

After more detailed discussions with DMTT it was agreed that PAA, PAI and DAC certificates shouldn't have expiration. I updated all test certs in this PR to have "No Well Defined Expiration" value.